docs/proposals/multiple-sources-for-applications.md

```diff
@@ -0,0 +1,225 @@
+    targetRevision: 8.5.8
+    helm:
+    valueFiles:
+        - values.yaml
+    chart: mysql
+  sources:                                          # new field
+    - repoURL: https://charts.bitnami.com/bitnami
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+      chart: mysql
+```
+
+### Add `externalValues` field in helm section of Application Spec
+
+Along with new `sources` field, add a new field for accepting external value files `externalValueFiles` under the `helm` section of each ApplicationSource.
+
+```yaml
+sources:
+    - repoURL: https://charts.bitnami.com/bitnami
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+        externalValueFiles:                         # new field
+          - repoURL: https://github.com/KaiReichart/argo-test-values.git
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
+```
+
+### Combined Example Application yaml
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: grafana
+  namespace: argocd
+spec:
+  destination:
+    namespace: monitoring
+    server: https://some.k8s.url.com:6443
+  project: default
+  source:
+    repoURL: https://charts.bitnami.com/bitnami
+    targetRevision: 8.5.8
+    helm:
+    valueFiles:
+        - values.yaml
+    chart: mysql
+  sources:                                          # new field
+    - repoURL: https://charts.bitnami.com/bitnami
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+        externalValueFiles:                         # new field
+          - repoURL: https://github.com/KaiReichart/argo-test-values.git
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
+    - repoURL: https://charts.bitnami.com/bitnami1
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+        externalValueFiles:
+          - repoURL: https://github.com/KaiReichart/argo-test-values.git
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
+  syncPolicy:
+    automated: {}
+```
+
+### Use cases
+
+Add a list of detailed use cases this enhancement intends to take care of.
+
+#### Use case 1: 
+As a user, I have an Application that uses the [elasticsearch](https://github.com/helm/charts/tree/master/incubator/elasticsearch) helm chart as source. Today, user needs to create a separate Application to configure the [elasticsearch-exporter](https://github.com/helm/charts/tree/master/stable/elasticsearch-exporter
+) to expose Prometheus metrics.
+https://github.com/argoproj/argo-cd/issues/677
+
+#### Use case 2: 
+We have a Helm Chart which is used in 30+ Services and each of them is customized for 3 possible environments.
+Replicating this Chart 30 times without a centralized Repo looks dirty. Can be a show stopper for the whole migration.
```

Comment:
Since this line comes from the [issue comments](https://github.com/argoproj/argo-cd/issues/2789#issuecomment-562495307), can we either rephrase this line or drop it into a quote block?